src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1796,17 +1797,75 @@ impl Cage {
         }
     }
 
-    //------------------------------------WRITE SYSCALL------------------------------------
-
+    /// ## ------------------WRITE SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `write_syscall()` attempts to write `count` bytes from the buffer
+    /// pointed to by `buf` to the file associated with the open file
+    /// descriptor, `fd`. The number of bytes written may be less than count
+    /// if, for example, there is insufficient space on the underlying
+    /// physical medium. On files that support seeking (for example, a
+    /// regular file), the write operation commences at the file offset, and
+    /// the file offset is incremented by the number of bytes written.
+    ///
```

Comment:
I assume that for files that do not support seeking it starts from zero?

Reply:
Adding that here would be great!

Reply:
Sockets, Pipes, and Epolls don't support seeking. Generally, for these files, the data is written differently, for Example: Sockets, we write data to the socket buffer which is passed over the network, for Pipes, data is written to a pipe buffer and processed in a FIFO manner.